### DISCUSSION TITLE
How to trigger binding action intentionally?

### DISCUSSION THREAD

**Discussion by Jager-yoo** [pointfreeco/swift-composable-architecture#1841] [Q&A] [answered]
Let's suppose that we have two `@BindableState` 'userID' and 'userEmail' with initialValues.
Those are connected to `TextField`s and the Reducer has:

```swift
case .binding(\.$userID):
  /* some local validations... */
  enum RequestUserNameID { }
  return env.checkUserName(state) // validates with server
    .map(Action.fetchCheckUserID)
    .cancellable(id: RequestUserNameID.self, cancelInFlight: true)

case .binding(\.$userEmail):
  /* some local validations... */
  enum RequestUserEmailID { }
  return env.checkUserEmail(state) // validates with server
    .map(Action.fetchCheckUserEmail)
    .cancellable(id: RequestUserEmailID.self, cancelInFlight: true)
```

I just want to trigger these binding actions intentionally, without modifying TextFields.
I tried `Effect(value:)` from another action case, but I'm not sure what to put as a binding argument. 🤷🏻 

<img width="706" alt="image" src="https://user-images.githubusercontent.com/71127966/212905907-453b2f66-2ac7-4b73-8fd4-94840a19359c.png">

Any ideas will be very appreciated. Thanks!

**Comment by tgrapperon** [comment, marked as answer]
Hey @Jager-yoo! It would be `.binding(.set(\.$userEmail, "something"))`, but you can maybe also refactor your domain to produce the same effect than the one from `case .binding(\.$userEmail)` using private functions. Re-emitting actions to reuse code is indeed inefficient, as you can read in the ["Performance"](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/performance#Sharing-logic-with-actions) article.

**Reply by Jager-yoo** [reply]
Hey! @tgrapperon, thank you so much!
The way you told me solved the problem I'm dealing with.
Also thanks for the performance article reference, really need to separate shared actions...
